src-electron/zcl/zcl-loader-silabs.js

```diff
@@ -2257,11 +2204,14 @@ async function loadSilabsZcl(db, metafile, isJson = false) {
     metadataFile: metafile,
     db: db,
   }
+  let isTransactionAlreadyExisting = dbApi.isTransactionActive()
   env.logDebug(`Loading Silabs zcl file: ${metafile}`)
   if (!fs.existsSync(metafile)) {
     throw new Error(`Can't locate: ${metafile}`)
   }
-  await dbApi.dbBeginTransaction(db)
+
+  if (!isTransactionAlreadyExisting) await dbApi.dbBeginTransaction(db)
```

Comment:
I guess we somehow keep pushing the transactions by 10s if one is going on. I am imagining a bunch of transactions just being pushed by 10s all the time and if that can also disrupt the order of things. for eg:
- transaction 1 is on going
- transaction 2 tries to execute but transaction 1 is ongoing so gets pushed by 10s
- transaction 3 tries to execute but transaction 1 is ongoing so gets pushed by 10s
- transaction 2 tries to execute again but transaction 1 is ongoing so gets pushed by another 10s
- transaction 1 finishes execution
- transaction 3 executes
- transaction 2 executes

In the above example now T3 happened before T2 even though the order of things was different. I am wondering if we should queue them to maintain the order of things instead of bumping them 10s every time.

Reply:
This is an issue if one transaction depends on another transaction to finish before it.

Reply:
Our transaction handling leaves a lot to be desired. The fact that SQLITE can run only one transaction at the time is a problem. What helps us, is that we need transactions ONLY when we're writing large bulk data into the database, which happens only in 3 occasions:
   1.) When loading a .zap file.
   2.) When loading a ZCL properties.
   3.) When loading a gen-templates.json

These used to be separated, but with this work, the case 1, can also automatically "include" cases 2 and 3.

The 10 second logic is a hopeful hack, essentially built as:
   1.) You load a.zap.
   2.) At the same time you load b.zap.
   
Loading of a file should be done in 10 seconds, so b.zap can wait until a.zap completes and closes transaction.

I can see tons of issues in this whole scenario, but honestly, I don't know what to do exactly about it. Any ideas appreciated.

Reply:
What if we change `isTransactionAlreadyExisting` to an array such as `AllIncomingTransactions = []` which can be an array of promises. We take them based on a queue. Execute them in the order they came in and remove them from the queue. I feel having an order for these transactions will make it easier for us to debug things if we ever run into issues and will probably help us avoid complex issues.

Reply:
Valid point. You should log an issue and we can implement that in a separate PR.

Reply:
Logged: https://github.com/project-chip/zap/issues/819
Assigned it to you.